docs/smallfile.md

```diff
@@ -0,0 +1,172 @@
+# Smallfile Benchmark
+
+[Smallfile](https://github.com/distributed-system-analysis/smallfile) is a python-based distributed POSIX workload generator which can be used to quickly measure performance for a variety of metadata-intensive workloads across an entire cluster.
+
+## Running Smallfile Benchmark using Ripsaw
+Once the operator has been installed following the instructions, one needs to modify the clients parameter(which is currently set to 0), to value greater than 0 in  [ripsaw_v1alpha1_smallfile_cr.yaml](../resources/crds/ripsaw_v1alpha1_smallfile_cr.yaml) to run default "create" the test. Also, in addtion to that, smallfile operator is completely dependent on storageclass and storagesize. Please make sure to double check the parameters in CRD file.
+
+```yaml
+apiVersion: ripsaw.cloudbulldozer.io/v1alpha1
+kind: Benchmark
+metadata:
+  name: example-benchmark
+  namespace: ripsaw
+spec:
+  workload:
+    name: smallfile
+    args:
+      clients: 1
+      operation: ["create"]
+      threads: 5
+      file_size: 64
+      files: 100
+      storageclass: rook-ceph-block # Provide if PV is needed
+      storagesize: 5Gi # Provide if PV is needed
+```
+
+Smallfile operator also gives the leverage to run multiple test operations in a user-defined sequence. Like in the [Custom Resource Definition file](../resources/crds/ripsaw_v1alpha1_smallfile_cr.yaml), the series of operation can be mentioned as:
+
+```yaml
+apiVersion: ripsaw.cloudbulldozer.io/v1alpha1
+kind: Benchmark
+metadata:
+  name: example-benchmark
+  namespace: ripsaw
+spec:
+  workload:
+    name: smallfile
+    args:
+      clients: 1
+      operation: ["create","read","append", "delete"]
+      threads: 5
+      file_size: 64
+      files: 100
+      storageclass: rook-ceph-block # Provide if PV is needed
+      storagesize: 5Gi # Provide if PV is needed
+```
+
+NOTE: While running the sequence of tests using smallfile workload, please make sure that the initial operation must be create, and the "delete" operation should come in termination, else smallfile might produce error due to meaningless sequence of tests. For example: 
+```bash
+operation: ["read","append","create", "delete"]
+#This will be meaningless sequence of test as trying to read something, which has not been created yet.The same logic applies for the append test as well. Hence, smallfile will produce error.
+```
+
+## Adding More options in smallfile tests
+
+Smallfile also comes with a variety of configurable options for running tests, following are the parameters that can be added to CR file:
+
+* response_times – if Y then save response time for each file operation in a
```

Comment:
Small nit, but could you bold the param names so they stand out?

Reply:
Agree with the idea. Will add that